stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -514,12 +514,38 @@ function (^)(A::AbstractMatrix{T}, p::Integer) where T<:Integer
 end
 function integerpow(A::AbstractMatrix{T}, p) where T
     TT = promote_op(^, T, typeof(p))
-    return (TT == T ? A : convert(AbstractMatrix{TT}, A))^Integer(p)
+    ATT = TT == T ? A : convert(AbstractMatrix{TT}, A)
+    return _integerpow(ATT, p)
 end
+_integerpow(A::AbstractMatrix, p) = A^Integer(p)
+function _integerpow(A::AbstractMatrix, p::Union{Float32, Float64})
+    # For these exponent types, not all values may be converted to Int
+    # We split the exponentiation into parts for which the exponent can be converted to an Int
+    if p < 0
+        return _integerpow(inv(A), -p)
+    end
+    m = typemax(Int)
```

Comment:
```suggestion
    m = prevpow(2, typemax(UInt))
```
`typemax(Int)` will not be accurate here (although arguably is not *that* far off). Note the inaccuracy of `divrem(2.0^64, typemax(Int)) == (2.0, 0.0)` due to the promotion `Int` -> `Float64`. Also, I believe the cost of `power_by_squaring` scales partly with the `count_ones` in the power.

Reply:
I think that the correct value here is `floatintmax(p)`

Reply:
I think you're right that it needs to be `<=maxintfloat(typeof(p))`, otherwise the `rem` could be wrong. So I think it needs to be `UInt(min(prevpow(2, typemax(UInt)), maxintfloat(typeof(p))))`.